chainerx_cc/examples/mnist_cc/mnist.cc

```diff
@@ -0,0 +1,81 @@
+#include "mnist.h"
+
+#include <algorithm>
+#include <cassert>
+#include <cstdint>
+#include <fstream>
+#include <memory>
+#include <string>
+
+#include "chainerx/array.h"
+#include "chainerx/device.h"
+#include "chainerx/dtype.h"
+#include "chainerx/routines/creation.h"
+#include "chainerx/shape.h"
+
+namespace {
+
+std::unique_ptr<char[]> ReadFile(const std::string& filename) {
```

Comment:
Why do you have to read the whole data first? I think it's not a good practice because it requires extra memory and you can't use the standard stream interfaces. It's also not good to access with the concrete offsets.

Reply:
Okay, I can change it back.

Reply:
Fixed.